rules/aip0203/field_behavior_required.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package aip0203
+
+import (
+	"fmt"
+
+	"bitbucket.org/creachadair/stringset"
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+)
+
+var minimumRequiredFieldBehavior = stringset.New(
+	"OPTIONAL", "REQUIRED", "OUTPUT_ONLY", "IMMUTABLE",
+)
+
+var fieldBehaviorRequired = &lint.MethodRule{
+	Name: lint.NewRuleName(203, "field-behavior-required"),
+	LintMethod: func(m *desc.MethodDescriptor) []lint.Problem {
+		// we only check requests, as OutputTypes are always
+		// OUTPUT_ONLY
+		it := m.GetInputType()
+		return checkFields(it)
+	},
+}
+
+func checkFields(m *desc.MessageDescriptor) []lint.Problem {
+	problems := []lint.Problem{}
+	for _, f := range m.GetFields() {
+		asMessage := f.GetMessageType()
+		if asMessage != nil {
+			problems = append(problems, checkFields(asMessage)...)
+		}
+		p := checkFieldBehavior(f)
+		if p != nil {
+			problems = append(problems, *p)
+		}
+	}
+	return problems
+}
+
+func checkFieldBehavior(f *desc.FieldDescriptor) *lint.Problem {
```

Comment:
Why return a reference and then indirect in the calling function. Recommend, changing this to return `lint.Problem` and then remove the `&` below and the `*` above.

Reply:
there could be no problems with the field behavior, note the nil at the end.

But if desirable I can just switch to a problem list like the others: this is a
little more versatile anyway.

Reply:
done, PTAL.